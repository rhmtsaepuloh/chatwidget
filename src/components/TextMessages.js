import React from "react";
import { MdDoneAll } from "react-icons/md";
import { urlify } from "../utils/messages";

const TextMessages = ({ message, timestamp }) => {
  return (
    <React.Fragment>
      <p
        dangerouslySetInnerHTML={{
          __html: `<p style='white-space: pre-wrap;overflow-wrap: break-word;margin: 0.25rem;'>${urlify(
            message
          )}</p>`,
        }}
      />
      <div className='file-timestamp flex items-center justify-end'>
        <p className='text-sm' style={{ color: "#979797" }}>
          {timestamp}
        </p>
        <MdDoneAll className='ml-1' />
      </div>
    </React.Fragment>
  );
};

export default TextMessages;
