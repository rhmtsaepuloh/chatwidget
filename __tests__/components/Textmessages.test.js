/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import TextMessages from "../../src/components/TextMessages";

const props = {
  message: "cek",
  timestamp: "10:21",
};

describe("render correctly text messages components", () => {
  test("renders", () => {
    const wrapper = shallow(<TextMessages {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
