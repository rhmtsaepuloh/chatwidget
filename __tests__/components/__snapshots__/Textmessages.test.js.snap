// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly text messages components renders 1`] = `ShallowWrapper {}`;
